LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/DefaultLoRaDataRequestHandler.cs

```diff
@@ -156,29 +177,28 @@ public async Task<LoRaDeviceRequestProcessResult> ProcessRequestAsync(LoRaReques
                         loRaDevice.UpdateRegion(request.Region.LoRaRegion, acceptChanges: false);
                 }
 
+                #region FrameCounterDown
                 // if deduplication already processed the next framecounter down, use that
                 var fcntDown = loRaADRResult?.FCntDown != null ? loRaADRResult.FCntDown : bundlerResult?.NextFCntDown;
 
-                if (fcntDown.HasValue)
-                {
-                    LogFrameCounterDownState(loRaDevice, fcntDown.Value);
-                }
```

Comment:
why was this removed?

Reply:
It was moved in `EnsureHasFcntDownAsync`: there we already check if we have a value for the fcntdown (0f11c3bce9a35e97b1cf3468a75b94202c6d06ef) 

Since `EnsureHasFcntDownAsync` is called from 3 other place, this change means we debug-log the frame counter down from those other places but I think a) it makes sense to have the same behavior in all these call sites b) that the logging is not too verbose. Let me know if you disagree.

Reply:
But wouldn't that mean, that we potentially log the value multiple times? Each time we call the Ensure method?

Reply:
Yes we do

> this change means we debug-log the frame counter down from those other places too but I think a) it makes sense to have the same behavior in all these call sites b) that the logging is not too verbose.

Do you find this a problem (log level is debug anyway)?

Reply:
as discussed. Let's make sure we don't emit the logs each time, but only when we really create it and have a valid counter.

Reply:
In 406c7d48724d6b1fb7136b324450f5c98c5ee289 we log if we already have a valid counter down, otherwise we log it the first time we get one from the Function. (we rely on the fact that comparing with null will always be false - [there is a compiler warning for this](https://docs.microsoft.com/en-us/dotnet/csharp/misc/cs0464))